checker.cpp

```diff
@@ -1,22 +1,55 @@
-#include <assert.h>
 #include <iostream>
+#include "checker.h"
+#ifdef UTEST
+#include "checker_test.h"
+#endif //UTEST
+
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
+bool InRange(float min, float max, float value){
+  if(value < min || value > max){
     return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
+  }
+  return true;
+}
+
+bool InRange(float max, float value){
+  if(value > max){
     return false;
   }
   return true;
 }
 
+bool temperatureCheck(float temperature){
+  bool inRange = InRange(minTemperature, maxTemperature, temperature);
+  if(!inRange) {
+    cout << "Temperature out of range!\n";
+  }
+  return inRange;
+}
+
+bool stateOfChargeCheck(float SOC){
```

Comment:
These functions return `bool`. From the name, it is unclear if they check and return true to mean 'all is ok', or true means 'action required'
Name boolean functions with some form of 'is'

Reply:
Duly noted, renamed methods.